ir/x86_intrinsics.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include "ir/instr.h"
+
+namespace IR {
+class X86IntrinBinOp final : public Instr {
+public:
+  enum Op {
+#define PROCESS(NAME, A, B, C, D, E, F) NAME,
+#include "x86_intrinsics_binop.inc"
+#undef PROCESS
+  };
+
+  // the shape of a vector is stored as <# of lanes, element bits>
```

Comment:
these should be removed from the header file and become static globals in the .cpp file. They are not needed here.

Reply:
Would you like to de-link the two different `shape_op0` definitions from the fact that they're associated with Binary and Ternary ops? If so, there will be a `shape_op0_binop` and shape_op0_terop` in the cpp, and the Binary and Ternary functions have to be changed to use them.

Reply:
You can give them different names so they don't clash. It's good to reduce the size of header files, as they control the compilation time. Also, this is internal stuff to the implementation of the class, doesn't have to be exposed to users.